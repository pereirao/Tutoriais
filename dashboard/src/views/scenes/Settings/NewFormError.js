// @flow

import React from 'react';
import _ from 'lodash';
import style_sheet from './Settings.module.css';

type NewFormErrorProps = { error_messages: Array<string> };
const NewFormError = ({error_messages}: NewFormErrorProps) => {
  if (_.isEmpty(error_messages)) return null;
  return (
    <div className={style_sheet.errorWrapper}>
      {error_messages.map((e, i) => <p key={`form_error${i}`}>{e}</p>)}
    </div>
  );
};

export default NewFormError;
