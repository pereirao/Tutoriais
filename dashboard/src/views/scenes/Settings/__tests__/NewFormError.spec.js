import React from 'react';
import NewFormError from '../NewFormError';

it('renders correctly', () => {
  expect(shallow(<NewFormError error_messages={['this is an error']} />)).toMatchSnapshot();
});
