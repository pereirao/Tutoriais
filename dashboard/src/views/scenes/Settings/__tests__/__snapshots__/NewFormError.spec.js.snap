// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="errorWrapper"
>
  <p>
    this is an error
  </p>
</div>
`;
